### PR TITLE
Add support for GCP's L4

### PR DESCRIPTION
**Pull request by hzeng-0**
This is together with a pull request on the skypilot repo: [link](https://github.com/skypilot-org/skypilot/pull/2212).

### CODE REVIEW

**Comment by infwinston** [comment]
Thanks @hzeng-0 !
How did you generated the L4 catalog? Another change we need to make is on `fetch_gcp.py` to make the catalog update automatic. because our github action will run this script to update it per 7 hours
https://github.com/skypilot-org/skypilot/blob/master/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py
https://github.com/skypilot-org/skypilot-catalog/actions/runs/5544136968/workflow#L52

**Comment by infwinston** [comment]
I think we probably don't need to touch `catalogs/v5/gcp/vms.csv` because `fetch_gcp.py` triggered by github action will update it?

File: catalogs/v5/gcp/images.csv [hunk 1]
```diff
@@ -2,4 +2,5 @@ Tag,Region,OS,OSVersion,ImageId,CreationDate
 skypilot:cpu-debian-10,,debian,10,projects/deeplearning-platform-release/global/images/common-cpu-v20221215,20221215
 skypilot:k80-debian-10,,debian,10,projects/deeplearning-platform-release/global/images/common-cu113-v20220701,20220701
 skypilot:gpu-debian-10,,debian,10,projects/deeplearning-platform-release/global/images/common-cu113-v20221215,20221215
+skypilot:l4-debian-11,,debian,11,projects/ml-images/global/images/c0-deeplearning-common-gpu-v20230615-debian-11-py310,20230615
```
Comment (infwinston) [inline catalogs/v5/gcp/images.csv:5]: do we know who maintain this image family `ml-images`? I'm a bit worried that the lifecycle of this image e.g., how long will they keep it or deprecate it?

**Review by infwinston — COMMENTED** [review]
thanks a lot! I only have one comment about the image.

File: catalogs/v5/gcp/images.csv (hunk 1 above)
Comment (infwinston) [reply catalogs/v5/gcp/images.csv:5]: ok I dug into this and found that this one may be more reliable
```
projects/deeplearning-platform-release/global/images/common-gpu-v20230615-debian-11-py310
```

Comment (hzeng-0) [reply catalogs/v5/gcp/images.csv:5]: I'm not sure who maintains the image family `ml-images`, so I think that should definitely be better! Let me make a commit

File: catalogs/v5/gcp/images.csv [hunk 2]
```diff
@@ -2,5 +2,5 @@ Tag,Region,OS,OSVersion,ImageId,CreationDate
 skypilot:cpu-debian-10,,debian,10,projects/deeplearning-platform-release/global/images/common-cpu-v20221215,20221215
 skypilot:k80-debian-10,,debian,10,projects/deeplearning-platform-release/global/images/common-cu113-v20220701,20220701
 skypilot:gpu-debian-10,,debian,10,projects/deeplearning-platform-release/global/images/common-cu113-v20221215,20221215
-skypilot:l4-debian-11,,debian,11,projects/ml-images/global/images/c0-deeplearning-common-gpu-v20230615-debian-11-py310,20230615
+skypilot:l4-debian-11,,debian,11,projects/deeplearning-platform-release/global/images/common-gpu-v20230615-debian-11-py310
```
Comment (infwinston) [inline catalogs/v5/gcp/images.csv:5]: missed `CreationDate`?

Comment (hzeng-0) [reply catalogs/v5/gcp/images.csv]: my bad, fixed

File: catalogs/v5/gcp/images.csv [hunk 3]
```diff
@@ -2,5 +2,5 @@ Tag,Region,OS,OSVersion,ImageId,CreationDate
 skypilot:cpu-debian-10,,debian,10,projects/deeplearning-platform-release/global/images/common-cpu-v20221215,20221215
 skypilot:k80-debian-10,,debian,10,projects/deeplearning-platform-release/global/images/common-cu113-v20220701,20220701
 skypilot:gpu-debian-10,,debian,10,projects/deeplearning-platform-release/global/images/common-cu113-v20221215,20221215
-skypilot:l4-debian-11,,debian,11,projects/deeplearning-platform-release/global/images/common-gpu-v20230615-debian-11-py310
+skypilot:l4-debian-11,,debian,11,projects/deeplearning-platform-release/global/images/common-gpu-v20230615-debian-11-py310,20230615
```
Comment (infwinston) [inline catalogs/v5/gcp/images.csv:5]: sorry let's probably call it `cuda118-debian-11` to be more precise?

Comment (hzeng-0) [reply catalogs/v5/gcp/images.csv:5]: sure, makes sense!

**Review by infwinston — APPROVED** [review]
lgtm

**Pull request merged**